modules/core/src/lib/layer.js

```diff
@@ -48,6 +49,11 @@ const TRACE_MATCHED = 'layer.matched';
 
 const EMPTY_ARRAY = Object.freeze([]);
 
+// Only compare the same two viewports once
+const areViewportsEqual = memoize(({oldViewport, viewport}) => {
```

Comment:
There are lots of viewports being created every cycle, I trust this flushes the cache after a while? 

Disclaimer: not a big fan of memoization generally, though i recognize it is sometimes and effective quick fix...

Reply:
This is holding onto at most 1 expired viewport instance.